scipy/optimize/_minimize.py

```diff
@@ -961,3 +972,42 @@ def standardize_constraints(constraints, x0, meth):
                 constraints.extend(old_constraints[1:])  # appends 1 if present
 
     return constraints
+
+
+def _optimize_result_for_equal_bounds(
+        fun, bounds, method, args=(), constraints=()
+):
+    """
+    Provides a default OptimizeResult for when a bounded minimization method
+    has (lb == ub).all().
+
+    Parameters
+    ----------
+    fun: callable
+    bounds: Bounds
+    method: str
+    constraints: Constraint
+    """
+    success = True
+    message = 'Optimization terminated successfully.'
+
+    # bounds is new-style
+    x0 = bounds.lb
+
+    if constraints:
+        constraints = standardize_constraints(constraints, x0, 'new')
+
+    for c in constraints:
+        maxcv = 0
+        pc = PreparedConstraint(c, x0)
+        violation = pc.violation(x0)
+        if np.sum(violation):
```

Comment:
I confirmed that `violation` is the absolute value of the violation of inequality (LT or GT) and equality constraints, so it's OK to sum. `violation` is a pretty handy function in this regard.

However, this is probably too strict of a check. We should probably be checking for violations exceeding a tolerance.

One could argue that if the user is going to fix all the decision variables, a very strict check like this is OK. On the other hand, consider if you were to run `res = minimize(...` on a constrained problem and get an optimal solution (`success = True`), then run it again with `bounds=Bounds(res.x, res.x)`. You might then get `success=False` because SLSQP doesn't satisfy equality constraints (or even inequality constraints, I don't think) exactly.

Reply:
> violation is a pretty handy function in this regard.

I had to add this for the constraints in `differential_evolution`.


I think this PR has to assume that `lb==ub` is going to result in `(res.x == lb).all()`, i.e. that the Bounds constraints would work perfectly, otherwise what would one return? For the other constraints what would a tolerance be? The magnitudes of the constraints may vary. It's not worth adding an extra parameter for what should be a rare occurrence.

Reply:
> that the Bounds constraints would work perfectly, otherwise what would one return?

Yes. I suggested `x=bounds.lb` above, too. No problem there.

> For the other constraints what would a tolerance be? 

That is actually what stopped me from tackling this on my own and posting an issue rather than just fixing it : P I knew we'd have to decide whether `constraints` were satisfied or not, and that would be a pain.

> The magnitudes of the constraints may vary.

That's why I suggested relative tolerance. Since you've selected to standardize the constraints in the new-style format, you can use the bound on the constraint as the denominator. For instance, if an upper bound constraint is violated, then you compare `violation/constraint.ub` against the tolerance - unless `constraint.ub=0`, in which case it's an absolute tolerance. I think that's about the best we can do without evaluating the constraint Jacobian or something.

(Unfortunately, this means we wouldn't be able to use the `violation` method as-is, because we don't know whether the upper of lower constraint is violated. Maybe add a private method to `PreparedConstraint` to calculate relative violation?)

> It's not worth adding an extra parameter for what should be a rare occurrence.

Sure, because it's so rare, I wouldn't suggest exposing an extra parameter. I think it would be forgivable to set our own (e.g. 1e-6 or 1e-8) or re-use `ftol`.

---

I suppose you could leave it as is, but if so, I would change the `message` to something like:

> All decision variables were fixed by `bounds`, but the decision variables do not satisfy the constraints exactly. (Maximum violation: {maxcv})."

I suppose it's fine as long as we are explicit in the message.